geomesa/src/main/scala/geotrellis/spark/io/geomesa/GeoMesaFeatureWriter.scala

```diff
@@ -0,0 +1,45 @@
+package geotrellis.spark.io.geomesa
+
+import geotrellis.spark._
+import geotrellis.vector._
+import geotrellis.geomesa.geotools._
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.geotools.data.Transaction
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.JavaConversions._
+
+class GeoMesaFeatureWriter(val instance: GeoMesaInstance)(implicit sc: SparkContext) extends Serializable {
+  def write[G <: Geometry, D: ? => Seq[(String, Any)]]
+    (layerId: LayerId, rdd: RDD[Feature[G, D]])
+    (implicit ev: Feature[G, D] => FeatureToGeoMesaSimpleFeatureMethods[G, D]): Unit = {
+
+    rdd
+      .map { f => val sf = f.toSimpleFeature(layerId.name); sf.getFeatureType -> sf }.groupByKey
+      .foreachPartition { (partition: Iterator[(SimpleFeatureType, Iterable[SimpleFeature])]) =>
+        // data store per partition
+        val dataStore = instance.accumuloDataStore
+        partition.foreach { case (sft, sf) =>
+          // register feature types and write features
+          dataStore.createSchema(sft)
+          val featureWriter = dataStore.getFeatureWriterAppend(sft.getTypeName, Transaction.AUTO_COMMIT)
+          val attrNames = featureWriter.getFeatureType.getAttributeDescriptors.map(_.getLocalName)
+          try {
+            sf.foreach { rawFeature =>
+              val newFeature = featureWriter.next()
+              attrNames.foreach(an => newFeature.setAttribute(an, rawFeature.getAttribute(an)))
```

Comment:
as a minor optimization, you could do
```
(0 until sft.getAttributeCount).foreach(i => newFeature.setAttribute(i, rawFeature.getAttribute(i))
```
lookup by index is always faster than lookup by attribute name